teps/0125-add-credential-filter-to-entrypoint-logger.md

```diff
@@ -0,0 +1,334 @@
+entrypoint. Also secrets printed via the `set -x` command are redacted from the final
+log output stream.
+
+### Requirements
+
+We do not want to give the entrypoint running in the pipeline step more/any permissions 
+on the API server. 
+It should not call the API server to get secret values for example.
+
+Operators should be able to enable/disable this behaviour via a feature flag in case any
+problems arise from this filtering mechanism.
+
+## Proposal
+
+The tekton controller looks for secret refs in volumes and env and also for CSI volumes 
+referencing secrets in the definition of the pipeline pod. It saves the information in 
+form of the names of environment variables and paths to files with secrets. The pod can
+then read those values and redact them.
+
+This design does not transmit actual secrets and the pod does not need access to the API 
+sever to retrieve them. It tells the pipeline pod only the places where it can find secrets
+attached to it. And those are the secrets that will be redacted. Because only those can be
+used in the corresponding pipeline step.
+
+The credential filter will redact values contained in those secrets from the output log stream
+and replace them with `[REDACTED:<secret-name>]`. Secret name is either the name of the 
+environment variable or the file path where the secret is stored in the pod.
+
+### Notes and Caveats
+
+Secrets can come from different sources than the one implemented in this filter.
+
+Therefore it is important to be very specific about what is being filtered.
+
+
+## Design Details
+
+While creating the pods for the pipeline, the controller will try to detect all secrets 
+attached to the pods. Usually secrets are attached to the pod either by setting environment 
+variables or mounting files into it. These detected secret locations are then transmitted 
+to the pipeline pod for credential filtering.
+
+### Secret Locations
+
+The secret locations detected by the controller are transmitted to the entrypoint as 
+an environment variable in json format with the following format:
+
+```json
+{
+  "environmentVariables": ["ENV1", "ENV2"],
+  "files": ["/path/to/secre1", "/path/to/secret2"]
+}
+```
+
+The entrypoint can then read the environment variables and file contents and redact them
+from the output log stream.
+
+### Secrets stored in Environment Variables
+
+The controller detects secrets stored as environment variables from the following pod syntax:
+
+```yaml
+env:
+- name: MY_SECRET_VALUE_IN_ENV
+  valueFrom:
+    secretKeyRef:
+      name: my-k8s-secret
+      key: secret_value_key
+```
+
+The secret key reference is the trigger to detect an environment variable that contains a secret
+and that needs to be redacted.
+
+### Secrets mounted as Files 
+
+Secrets can also be mounted into pods via files in different ways. The following pod
+syntax should be supported in the secret detection logic.
+
+```yaml
+volumes:
+- name: secret-volume
+  secret:
+    secretName: my-k8s-secret
```

Comment:
Thanks for including this example-- I was a bit concerned that the PR wouldn't work for workspaces w/ secrets as workspace bindings but based on this I think it will. We should just make sure this case is covered by tests.

Reply:
I have just added this to the test case that I created